### DISCUSSION TITLE
Color Averaging

### DISCUSSION THREAD

**Discussion by CaliOn2** [YaLTeR/niri#505] [General]
the way colors are averaged when borders have 2 bright different colors
(for example 0xff0000 (red) and 0x00ff00 (green)) the gradient colors in the middle are very inaccurately dark and ruin some color combinations.

**Comment by YaLTeR** [comment]
The current algorithm matches what I get in browsers with CSS linear-gradient, and is documented as such. So maybe it should be a separate property?

**Reply by kklub123** [reply]
you mean like a config option?

**Reply by YaLTeR** [reply]
Specified in some way in the config

**Reply by CaliOn2** [reply]
@YaLTeR I've updated it so that the default is css-linear and there is a config option for true linear i plan to add more color gradient options in the next few days( 4 example Lch and Oklab)
give me a heads up if this meets requirements for a merge or if you have any other gradient requests
also if you do merge this please take a good look at the code first as i've made a drastic change in the shaders file structure to allow the integration of a seperate gradients file. Cheers!
